dts/bindings/sensor/st,lps22df-common.yaml

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2023 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+include: sensor-device.yaml
+
+properties:
+  drdy-gpios:
+    type: phandle-array
+    description: |
+      DRDY pin
+
+      This pin defaults to active high when produced by the sensor.
+      The property value should ensure the flags properly describe
+      the signal that is presented to the driver.
+
+  drdy-pulsed:
+    type: boolean
+    description: |
+      Selects the pulsed mode for data-ready interrupt when enabled,
+      and the latched mode when disabled.
+
+  odr:
+    type: int
+    default: 0
+    description: |
+        Specify the output data rate expressed in samples per second (Hz).
+        The default is the power-on reset value.
+
+        0 = Power-Down
+        1 = 1Hz
+        2 = 4Hz
+        3 = 10Hz
+        4 = 25Hz
+        5 = 50Hz
+        6 = 75Hz
+        7 = 100Hz
+        8 = 200Hz
+    enum:
+      - 0
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+      - 6
+      - 7
+      - 8
```

Comment:
not for this PR since this is coherent with other ST drivers, but if you or anyone from ST feels like doing a swipe of the other ST bindings in the future: I think it'd be a nice usability improvements to have a matching set of `#define`s for these in `include/zephyr/dt-bindings/sensor`, one file per class of device I guess.

Then the dt instance of these would look something like `odr = <LPS22DF_ODR_100HZ>;` rather than `odr = <7>;`, one would not have to cross reference the binding documentation, you could change the description of this to just say "any of LPS22DF_ODR_..." etc...

Reply:
> not for this PR since this is coherent with other ST drivers, but if you or anyone from ST feels like doing a swipe of the other ST bindings in the future: I think it'd be a nice usability improvements to have a matching set of `#define`s for these in `include/zephyr/dt-bindings/sensor`, one file per class of device I guess.
> 

Sure, I can think about it. Probably I would be the person that would do it, and I can do a pass for all ST devices in a single PR.

> Then the dt instance of these would look something like `odr = <LPS22DF_ODR_100HZ>;` rather than `odr = <7>;`, one would not have to cross reference the binding documentation, you could change the description of this to just say "any of LPS22DF_ODR_..." etc...

Exactly. At that point there would be no need for explain each case one by one, as the name itself would describe it. And to be hones it is very bad to read now.

Reply:
Yeah, cross referencing is painful